x-pack/legacy/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/create_analytics_form/create_analytics_form.tsx

```diff
@@ -38,11 +37,15 @@ import { Messages } from './messages';
 import { JobType } from './job_type';
 import { JobDescriptionInput } from './job_description';
 import { mmlUnitInvalidErrorMessage } from '../../hooks/use_create_analytics_form/reducer';
+import {
+  IndexPattern,
+  indexPatterns,
+} from '../../../../../../../../../../../src/plugins/data/public';
 
 // based on code used by `ui/index_patterns` internally
 // remove the space character from the list of illegal characters
-INDEX_PATTERN_ILLEGAL_CHARACTERS.pop();
-const characterList = INDEX_PATTERN_ILLEGAL_CHARACTERS.join(', ');
+indexPatterns.ILLEGAL_CHARACTERS.pop();
```

Comment:
Are we sure we want to allow this type of mutation? I know that it was there but I'm not sure if we want to allow it. Also the above commit that use the old `ui/index_patterns` folder should change

Reply:
Or maybe instead of using `ILLEGAL_CHARACTERS` just use `ILLEGAL_CHARACTERS_VISIBLE` that doesn't contain the space

Reply:
cc @peteharverson

Reply:
Yes to avoid the mutation, could do
```
const illegalCharacters = [...INDEX_PATTERN_ILLEGAL_CHARACTERS];
illegalCharacters.pop();
const characterList = illegalCharacters.join(', ');
```
and the 
```
based on code used by ui/index_patterns internally
```
comment, could be edited to just `based on code used by indexPatterns internally` to remove the reference to the old `ui_index_patterns` folder.

Reply:
@markov00 I agree with you, we cannot leave it as it is. Looks like you are absolutely right and here we should use `ILLEGAL_CHARACTERS_VISIBLE` 

@liza-mae  @peteharverson  I've updated this PR please have a look it